muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -0,0 +1,171 @@
+package fi.otavanopisto.muikku.plugins.notes;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.ejb.Stateful;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import fi.otavanopisto.muikku.plugin.PluginRESTService;
+import fi.otavanopisto.muikku.plugins.notes.model.Note;
+import fi.otavanopisto.muikku.schooldata.RestCatchSchoolDataExceptions;
+import fi.otavanopisto.muikku.schooldata.SchoolDataIdentifier;
+import fi.otavanopisto.muikku.session.SessionController;
+import fi.otavanopisto.security.rest.RESTPermit;
+import fi.otavanopisto.security.rest.RESTPermit.Handling;
+
+@RequestScoped
+@Stateful
+@Produces("application/json")
+@Path ("/notes")
+@RestCatchSchoolDataExceptions
+public class NotesRESTService extends PluginRESTService {
+
+  private static final long serialVersionUID = 6610657511716011677L;
+  
+  @Inject
+  private Logger logger;
+  
+  @Inject
+  private SessionController sessionController;
+  
+  @Inject
+  private NotesController notesController;
+  
+
+  // mApi() call (mApi().notes.note.create(noteRestModel)
+  //
+  // noteRestModel: = {
+  //  title: String, 
+  //  description: String, 
+  //  type: emun MANUAL
+  //  priority: enum LOW/NORMAL/HIGH, 
+  //  pinned: Boolean, 
+  //  owner: userIdentifier,
+  //  creator: loggedUser
+  //  };
+  @POST
+  @Path("/note")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response createNote(NoteRestModel note) {
+    
+    if (!sessionController.isLoggedIn()) {
+      return Response.status(Status.UNAUTHORIZED).build();
+    }
+    
+    if (!sessionController.hasEnvironmentPermission(NotesPermissions.NOTES_CREATE)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    try {
```

Comment:
No need for `try..catch`. If the creation fails, the end result will be a stacktrace and 500 (Internal server error) anyway. Of course, the endpoint could first validate all mandatory values in the payload object something like...

```
if (StringUtils.isEmpty(note.getTitle()) {
  return Response.status(Status.BAD_REQUEST).entity("Missing title").build();
}
```

...but that might be overkill as there are "a ton" of mandatory fields. Just telling the UI people which values are required should be sufficient.